TYPO3.Neos/Resources/Public/JavaScript/Content/Components/ContentContextBar.js

```diff
@@ -2,38 +2,44 @@
  * Context context bar
  */
 define(
-[
-	'emberjs',
-	'./ContextBar',
-	'./ContentDimensionSelector',
-	'../FullScreenController',
-	'../Application',
-	'Shared/Configuration',
-	'text!./ContentContextBar.html'
-], function(
-	Ember,
-	ContextBar,
-	ContentDimensionSelector,
-	FullScreenController,
-	ContentModule,
-	Configuration,
-	template
-) {
-	return ContextBar.extend({
-		classNames: 'neos-content-context-bar',
-		template: Ember.Handlebars.compile(template),
-		ContentDimensionSelector: ContentDimensionSelector,
-		fullScreenController: FullScreenController,
-		Configuration: Configuration,
-		init: function() {
-			this.updateCurrentUri();
-			var that = this;
-			ContentModule.on('pageLoaded', function() {
-				that.updateCurrentUri();
-			});
-		},
-		updateCurrentUri: function() {
-			this.set('liveUri', location.href.replace(/@[A-Za-z0-9;&,\-_=]+/g, ''));
-		}
-	});
-});
\ No newline at end of file
+  [
+    'emberjs',
+    './ContextBar',
+    './ContentDimensionSelector',
+    './ContentDimensionController',
+    '../FullScreenController',
+    '../Application',
+    './TargetWorkspaceController',
+    'Shared/Configuration',
+    'text!./ContentContextBar.html'
+  ], function (Ember,
+               ContextBar,
+               ContentDimensionSelector,
+               ContentDimensionController,
+               FullScreenController,
+               ContentModule,
+               TargetWorkspaceController,
+               Configuration,
+               template) {
+    return ContextBar.extend({
+      classNames: 'neos-content-context-bar',
+      template: Ember.Handlebars.compile(template),
+      ContentDimensionSelector: ContentDimensionSelector,
+      contentDimensionController: ContentDimensionController,
+      fullScreenController: FullScreenController,
+      targetWorkspaceController: TargetWorkspaceController,
+      Configuration: Configuration,
+
+      /**
+       * Computed property of preview uri
+       */
+      previewUri: function() {
+        var targetWorkspaceName = this.get('targetWorkspaceController.targetWorkspace.name');
+        if (targetWorkspaceName === 'live') {
+          return location.href.replace(/@[A-Za-z0-9;&,\-_=]+/g, '');
+        } else {
+          return location.href.replace(/@[A-Za-z0-9;&,\-_=]+/g, '@' + targetWorkspaceName);
+        }
+      }.property('targetWorkspaceController.targetWorkspace', 'contentDimensionController.selectedDimensions')
```

Comment:
with this change it doesn't update the URL when changing page as it needs to and did before as far as I can tell..

since there's no state object, you need to listen to the pageLoaded event and trigger it from there.. either by adding another dependent property or using ``this.get('targetWorkspaceController ').notifyPropertyChange('activeNode');``.. both approaches kinda suck though